core/string/ustring.cpp

```diff
@@ -4352,9 +4352,22 @@ bool String::is_valid_filename() const {
 		return false;
 	}
 
+	// TODO: Use invalid_filename_characters
 	return !(find(":") != -1 || find("/") != -1 || find("\\") != -1 || find("?") != -1 || find("*") != -1 || find("\"") != -1 || find("|") != -1 || find("%") != -1 || find("<") != -1 || find(">") != -1);
 }
 
+// Changes made to the set of invalid characters must also be reflected in the String documentation and the is_valid_filename function.
+static const char *invalid_filename_characters = ": / \\ ? * \" | % < >";
+
+String String::validate_filename() const {
+	Vector<String> chars = String(invalid_filename_characters).split(" ");
```

Comment:
Could be `LocalVector` maybe.

Reply:
The issue is .split() allocates a Vector. `validate_node_name` does the same. It's called infrequently in imoprt code.
I was originally thinking of doing .replace().replace()...

Reply:
It can be for a follow-up PR if needed/relevant, but I guess `invalid_filename_characters` could be made a static `LocalVector<CharString>` with all the chars already split, so we don't need to redo this computation for what's going to always be the same every time we validate a filename.